synapse/config/emailconfig.py

```diff
@@ -74,19 +74,20 @@ def read_config(self, config, **kwargs):
             "renew_at"
         )
 
-        email_trust_identity_server_for_password_resets = email_config.get(
-            "trust_identity_server_for_password_resets", False
```

Comment:
Could we give some help to people who set `trust_identity_server_for_password_resets` ? Generally we try very hard not to break existing configs, and breaking them silently is worse yet. Perhaps throw a ConfigError if `trust_identity_server_for_password_resets` is set?